src/Compiler/Service/ServiceStructure.fs

```diff
@@ -717,6 +717,49 @@ module Structure =
                 match directives with
                 | [] -> ()
                 | ConditionalDirectiveTrivia.If _ as ifDirective :: directives -> group directives (ifDirective :: stack) sourceLines
+                | ConditionalDirectiveTrivia.Elif(_, elifRange) as elifDirective :: directives ->
+                    match stack with
+                    | ConditionalDirectiveTrivia.If(_, ifRange) :: stack ->
+                        let startLineIndex = elifRange.StartLine - 2
+
+                        if startLineIndex >= 0 then
+                            // start of #if until the end of the line directly above #elif
+                            let range =
+                                mkFileIndexRange
+                                    ifRange.FileIndex
+                                    ifRange.Start
+                                    (mkPos (elifRange.StartLine - 1) sourceLines[startLineIndex].Length)
+
+                            {
+                                Scope = Scope.HashDirective
+                                Collapse = Collapse.Same
+                                Range = range
+                                CollapseRange = range
+                            }
+                            |> acc.Add
+
+                        group directives (elifDirective :: stack) sourceLines
+                    | ConditionalDirectiveTrivia.Elif(_, prevElifRange) :: stack ->
+                        let startLineIndex = elifRange.StartLine - 2
+
+                        if startLineIndex >= 0 then
+                            // start of previous #elif until the end of the line directly above current #elif
+                            let range =
+                                mkFileIndexRange
+                                    prevElifRange.FileIndex
+                                    prevElifRange.Start
+                                    (mkPos (elifRange.StartLine - 1) sourceLines[startLineIndex].Length)
+
+                            {
+                                Scope = Scope.HashDirective
+                                Collapse = Collapse.Same
+                                Range = range
+                                CollapseRange = range
+                            }
+                            |> acc.Add
```

Comment:
This code seems to duplicate the most of the previous branch. Could you refactor it, please?